src/app/add-item/add-item.component.ts

```diff
@@ -58,11 +72,22 @@ export class AddItemComponent implements OnInit {
     }
     const newItem: AddActionItem = {
       title: title.value,
-      projectName: projectName.value.name,
+      projectName: projectName.value,
       type: 'General',
       completed: '0',
       dueDate: dueDate.value
     };
     return newItem;
   }
+  editItem(): void {
+    this.isEditingActionItem = true;
+    const editedItemTitle = this.dialogForm.get('name')!.value;
+    const editedItemProject = this.dialogForm.get('project')!.value;
+    const editedItemDueDate = this.dialogForm.get('dueDate')!.value;
```

Comment:
**FIX:** You can reuse item preparation code from the create action item - this should be the same.